concordium-sdk/src/main/java/com/concordium/sdk/transactions/Transaction.java

```diff
@@ -51,4 +51,23 @@ static void verifyTransferInput(AccountAddress sender, AccountNonce nonce, Expir
             throw TransactionCreationException.from(new IllegalArgumentException("Signer cannot be null or empty"));
         }
     }
+    
+    static void verifyRegisterDataInput(AccountAddress sender, AccountNonce nonce, Expiry expiry, Data data, TransactionSigner signer) throws TransactionCreationException {
```

Comment:
I know that `verifyTransferInput` is currently present in `Transaction.java` however this is for historic reasons as there were only `Transfer` and `TransferWithMemo` before.

I think now is the time to factor these verification functions away from `Transaction.java` as it will quickly ramp up.

Reply:
Refactored and moved the method to Builder.